flytekit/deck/renderer.py

```diff
@@ -86,3 +86,77 @@ def to_html(self, source_code: str) -> str:
         css = formatter.get_style_defs(".highlight").replace("#fff0f0", "#ffffff")
         html = highlight(source_code, PythonLexer(), formatter)
         return f"<style>{css}</style>{html}"
+
+
+class PythonDependencyRenderer:
+    """
+    PythonDependencyDeck is a deck that contains information about packages installed via pip.
+    """
+
+    def __init__(self, title: str = "Python Dependencies"):
+        self._title = title
+
+    def to_html(self) -> str:
+        import json
+        import subprocess
+        import sys
+
+        from flytekit.loggers import logger
+
+        try:
+            installed_packages = json.loads(
+                subprocess.check_output([sys.executable, "-m", "pip", "list", "--format", "json"])
+            )
+            requirements_txt = (
+                subprocess.check_output([sys.executable, "-m", "pip", "freeze"])
+                .decode("utf-8")
+                .replace("\\n", "\n")
+                .rstrip()
+            )
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Error occurred while fetching installed packages: {e}")
+            return ""
+
+        table = (
+            "<table>\n<tr>\n<th style='text-align:left;'>Name</th>\n<th style='text-align:left;'>Version</th>\n</tr>\n"
+        )
+
+        for entry in installed_packages:
+            table += f"<tr>\n<td>{entry['name']}</td>\n<td>{entry['version']}</td>\n</tr>\n"
+
+        table += "</table>"
+
+        html = f"""
+        <!DOCTYPE html>
+        <html lang="en">
+        <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Flyte Dependencies</title>
+        <script>
+        async function copyTable() {{
+          var requirements_txt = document.getElementById('requirements_txt');
+
+          try {{
+            await navigator.clipboard.writeText(requirements_txt.innerText);
+          }} catch (err) {{
+            console.log('Error accessing the clipboard: ' + err);
+          }}
+        }}
+        </script>
+        </head>
+        <body>
+
+        <button onclick="copyTable()">
```

Comment:
@eapolinario This copy button does not work. I get this error in the console:

![Screenshot 2024-04-02 at 3 14 42 PM](https://github.com/flyteorg/flytekit/assets/5402633/c7572579-8208-4030-ae43-df6477be3f8b)

![Screenshot 2024-04-02 at 3 15 38 PM](https://github.com/flyteorg/flytekit/assets/5402633/a71456aa-6ccf-45f4-9f27-dec8e97657b0)

If we can not get this to work, I'm okay with removing the copy button and just showing the table.

Reply:
This is blocked on https://github.com/flyteorg/flyteconsole/pull/852.

Reply:
That just got merged, so next Flyte release is going to have this.